friends/src/components/EachFriend.js

```diff
@@ -0,0 +1,142 @@
+import React from "react";
+import styled from "styled-components";
+
+const FriendsContainerStyle = styled.div`
+  border: 1px solid black;
+  margin: 20px auto;
+  width: 500px;
+  .edit-button {
+    width: 80px;
+    height: 40px;
+    &:hover {
+      background: darkgray;
+    }
+  }
+  .delete-button {
+    margin-left: 20px;
+    width: 80px;
+    height: 40px;
+    background: indianred;
+    &:hover {
+      background: red;
+    }
+  }
+`;
+
+const FormStyle = styled.div`
+  border: 1px solid black;
+  width: 500px;
+  margin: 0 auto;
+  button {
+    margin: 20px;
+    width: 100px;
+    height: 40px;
+    background: lawngreen;
+    border-radius: 10px;
+    &:hover {
+      background: green;
+    }
+  }
+`;
+
+class EachFriend extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      edit: false,
+      currentName: "",
+      currentAge: "",
+      currentEmail: ""
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      currentName: this.props.friend.name,
+      currentAge: this.props.friend.age,
+      currentEmail: this.props.friend.email
+    });
+  }
+
+  inputHandler = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  editHandlerTrue = () => {
+    this.setState({ edit: true });
+  };
+
+  editHandlerFalse = (id, name, age, email) => {
+    this.props.editPost(id, name, age, email);
+    this.setState({ edit: false });
+  };
+
+  render() {
```

Comment:
For a DRYer implementation...
```suggestion
  render() {
  const {id, name, age, email} = this.props.friend;
```